File: Leetcode/Python/2379. Minimum Recolors to Get K Consecutive Black Blocks/solution.py
# https://leetcode.com/problems/minimum-recolors-to-get-k-consecutive-black-blocks/description

class Solution:
    def minimumRecolors(self, blocks: str, k: int) -> int:
        current_recolor = 0

        # First k block
        for idx in range(k):
            if blocks[idx] == "W":
                current_recolor += 1

        min_recolor = min(min_recolor, current_recolor)

        for idx in range(k, len(blocks)):
            if blocks[idx] == "W":
                current_recolor += 1
            if blocks[idx - k] == "W":
                current_recolor -= 1

            min_recolor = min(min_recolor, current_recolor)

        return min_recolor

File: Leetcode/Typescript/2379. Minimum Recolors to Get K Consecutive Black Blocks/solution.ts
// https://leetcode.com/problems/minimum-recolors-to-get-k-consecutive-black-blocks/description

function minimumRecolors(blocks: string, k: number): number {
  let minRecolor = k;
  let needRecolor = k;

  for (let idx = 0; idx < blocks.length; idx++) {
    if (blocks[idx] === "B") {
      needRecolor -= 1;
    }

    if (idx >= k) {
      if (blocks[idx - k] === "B") {
        needRecolor += 1;
      }
    }
    minRecolor = Math.min(minRecolor, needRecolor);
  }

  return minRecolor;
}
